File: 9/C++/Solution.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
  public:
	int getResult(int n, int m)
	{
		int last = 0;
		for (int i = 2; i <= n; i++)
		{
			last = (last + m) % i;
		}
		// 实际编号为(1~n)
		return last + 1;
	}
};

int main()
{
	Solution solution;
	cout << solution.getResult(5, 3) << endl;

	return 0;
}

File: 9/Python/Solution.py
# -*- coding:utf-8 -*-


class Solution:
    def getResult(self, n, m):
        l = [i for i in range(n)]
        p = 0
        while len(l) != 1:
            p = (p + m - 1) % len(l)
            l.pop(p)
        return l[0] + 1


print(Solution().getResult(5, 3))
